client/components/LearnerViewDirective.js

```diff
@@ -410,11 +469,21 @@ tie.directive('learnerView', [function() {
           loadQuestion(questionId, questionSet.getIntroductionParagraphs());
         };
 
-        $scope.submitCode = function(code) {
+        var callSolutionHandlerService = function() {
+          var code = this.code;
           SolutionHandlerService.processSolutionAsync(
-            prompts[currentPromptIndex], code,
-            question.getAuxiliaryCode(language), language
-          ).then(setFeedback);
+              prompts[currentPromptIndex], code,
+              question.getAuxiliaryCode(language), language
+              ).then(setFeedback);
+        };
+
+        $scope.submitCode = function(code) {
```

Comment:
Not sure you need to break the inner thing out into a separate function -- it just adds indirection. Could you just do something like this?

        $scope.submitCode = function(code) {
          $scope.hideDotDiv = false;
          feedbackDiv.scrollTop = feedbackDiv.scrollHeight + 17;
          $timeout(function() {
            SolutionHandlerService.processSolutionAsync(
              prompts[currentPromptIndex], code,
              question.getAuxiliaryCode(language), language
            ).then(setFeedback);
          }, 0);
        };